1_fetch/src/fetch_precip_spatial.R

```diff
@@ -0,0 +1,50 @@
+
+#' @param ind_file scipiper indicator file
+#' @param view_polygon from the view_polygon target
+#' and `bbox` e.g. [-83, 25, -80, 28]
+fetch_precip_spatial <- function(ind_file, view_polygon) {
+
+  nc <- ncdf4::nc_open("https://cida.usgs.gov/thredds/dodsC/stageiv_combined")
+
+  lon <- ncdf4::ncvar_get(nc, "lon")
+  lat <- ncdf4::ncvar_get(nc, "lat")
+
+  x <- matrix(rep(c(1:ncol(lon)), nrow(lon)),
+              nrow = nrow(lon), ncol = ncol(lon),
+              byrow = TRUE)
+
+  y <- matrix(rep(c(1:nrow(lon)), ncol(lon)),
+              nrow = nrow(lon), ncol = ncol(lon),
+              byrow = FALSE)
+
+  sf_points <- data.frame(x = matrix(x, ncol = 1),
+                          y = matrix(y, ncol = 1),
+                          lon = matrix(lon, ncol = 1),
+                          lat = matrix(lat, ncol = 1))
+
+  sf_points <- sf::st_as_sf(sf_points,
+                            coords = c("lon", "lat"),
+                            crs = "+init=epsg:4326",
+                            agr = "constant")
+
+  sf_points_sub <- sf::st_intersection(
+    sf::st_transform(sf_points,
+                     sf::st_crs(view_polygon)),
+    view_polygon)
```

Comment:
Hmm, even w/ `view_polygon`, I seem to be seeing only a subset of the domain: 
![image](https://user-images.githubusercontent.com/2349007/42510469-985484b6-8414-11e8-8c29-d10db0261341.png)